src/main/java/network/brightspots/rcv/Tabulator.java

```diff
@@ -548,14 +551,21 @@ private List<String> identifyWinners(
       // * If this is a single-winner election in which it's possible for no candidate to reach the
       //   threshold (i.e. "first round determines threshold" is set), the tiebreaker will choose
       //   the only winner.
-      boolean useTiebreakerIfNeeded = config.isMultiSeatAllowOnlyOneWinnerPerRoundEnabled()
-          || config.isFirstRoundDeterminesThresholdEnabled();
-      if (useTiebreakerIfNeeded && selectedWinners.size() > 1) {
+      boolean needsTiebreakMultipleWinners = selectedWinners.size() > 1
+          && (config.isMultiSeatAllowOnlyOneWinnerPerRoundEnabled()
+              || config.isFirstRoundDeterminesThresholdEnabled());
+      // Edge case: there are two candidates remaining, and we are not allowed to go down to
+      // one candidate in the final round. Break the tie with an election here, instead of
+      // an elimination later.
+      boolean needsTiebreakNoWinners = selectedWinners.size() == 0
+          && (currentRoundTally.numActiveCandidates() == 2
+              && config.doPreventOneCandidateInFinalRound());
+      if (needsTiebreakMultipleWinners || needsTiebreakNoWinners) {
         // currentRoundTallyToCandidates is sorted from low to high, so just look at the last key
         BigDecimal maxVotes = currentRoundTallyToCandidates.lastKey();
         selectedWinners = currentRoundTallyToCandidates.get(maxVotes);
         // But if there are multiple candidates tied for the max tally, we need to break the tie.
-        if (selectedWinners.size() > 1) {
+        if (selectedWinners.size() > 1 || (selectedWinners.size() == 0)) {
```

Comment:
nit: extraneous parens around second half of this

Reply:
But also, will `selectedWinners` ever be empty on this line? The previous statement just set it to `currentRoundTallyToCandidates.get(maxVotes)`.

I think this section of the code would be easier to follow if you kept the checks for the different scenarios distinct and then ended with `if (needTiebreak) { doTiebreak }`